qa/restart-upgrade/build.gradle

```diff
@@ -0,0 +1,87 @@
+/*
+ *  Copyright OpenSearch Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import org.opensearch.gradle.testclusters.StandaloneRestIntegTestTask
+import org.apache.tools.ant.taskdefs.condition.Os
+import java.util.concurrent.Callable
+
+apply from : "$rootDir/qa/build.gradle"
+
+String default_bwc_version = System.getProperty("bwc.version")
+String neural_search_bwc_version = System.getProperty("tests.bwc.version", default_bwc_version)
+boolean isSnapshot = neural_search_bwc_version.contains("-SNAPSHOT")
+String baseName = "neuralSearchBwcCluster-restart"
+String knn_bwc_version_no_qualifier = isSnapshot ? neural_search_bwc_version - "-SNAPSHOT" : neural_search_bwc_version
+
+List<Provider<RegularFile>> plugins = [provider(new Callable<RegularFile>(){
+    @Override
+    RegularFile call() throws Exception {
+        return new RegularFile() {
+            @Override
+            File getAsFile() {
+                return configurations.zipArchive.asFileTree.matching{include "**/opensearch-ml-plugin-${opensearch_build}.zip"}.getSingleFile()
+            }
+        }
+    }
+}), provider(new Callable<RegularFile>(){
+    @Override
+    RegularFile call() throws Exception {
+        return new RegularFile() {
+            @Override
+            File getAsFile() {
+                return configurations.zipArchive.asFileTree.matching{include "**/opensearch-knn-${opensearch_build}.zip"}.getSingleFile()
+            }
+        }
+    }
+}), rootProject.tasks.bundlePlugin.archiveFile]
+
+// Creates a test cluster of previous version and loads k-NN plugin of bwcVersion
+testClusters {
+    "${baseName}" {
+        testDistribution = "ARCHIVE"
+        versions = [neural_search_bwc_version, opensearch_version]
+        numberOfNodes = 3
+        jvmArgs("-Xms1g", "-Xmx4g")
```

Comment:
Why these args?

Reply:
There is an issue with ml-commons. It sends Circuit breaker exceptions due to out of memory. This has been given so that nodes can change memory requirements on runtime. This issue is consistent with integ tests too. So @martin-gaievski made a change [here](https://github.com/opensearch-project/neural-search/blob/5daddfd2f10486ad4688742aa26db82461b6a73c/build.gradle#L218)